File: src/SystemManager.cpp
#include "SystemManager.h"

void SystemManager::registerComponent(BaseComponent* component) {
    components.push_back(component);  // Add the new component to the list of registered components
}

void SystemManager::parseData() {
    for (auto component : components) {
        component->parseData();  // Call the parseData method for each registered component
    }
}

File: python/src/system_manager.py
class SystemManager:
    def __init__(self):
        self.components = []

    def register_component(self, component):
        self.components.append(component)

    def parse_data(self):
        for component in self.components:
            component.parse_data()

    def print_components(self, log_stream):
        for component in self.components:
            component.print_component(log_stream)
